src/bindgen/library.rs

```diff
@@ -443,4 +450,53 @@ impl Library {
             x.mangle_paths(&monomorphs);
         }
     }
+
+    // Apply associated type replacement to everything
+    fn replace_assoc_types(&mut self){
+
+        let assoc_map = &self.assoc_types;
+
+        // Replace associated types in function signatures
+        for func in self.functions.iter_mut(){
+            // Check return type
+            if let Some(ty) = find_concrete(&func.ret, assoc_map){
+                func.ret = ty
+            };
```

Comment:
I think you have to do a recursive replacement for all of replacements (function args, structs, consts, etc)?

Reply:
also cover with tests

Reply:
Done. Tested on 1 layer of recursion